Description: If the profile specified isn't found then try to use the default one.
Author: Iain Lane <iain.lane@canonical.com>
Forwarded: https://bugzilla.gnome.org/show_bug.cgi?id=747106

Index: b/src/terminal-options.c
===================================================================
--- a/src/terminal-options.c
+++ b/src/terminal-options.c
@@ -321,7 +321,16 @@
   profile = terminal_profiles_list_dup_uuid_or_name (terminal_options_ensure_profiles_list (options),
                                                      value, error);
   if (profile == NULL)
-    return FALSE;
+  {
+      g_warning ("Profile '%s' specified but not found. Attempting to fall back "
+              "to the default profile.", value);
+      g_clear_error (error);
+      profile = terminal_profiles_list_dup_uuid_or_name (terminal_options_ensure_profiles_list (options),
+                                                         NULL, error);
+  }
+
+  if (profile == NULL)
+      return FALSE;
 
   if (options->initial_windows)
     {
@@ -381,6 +390,16 @@
 
   profile = terminal_profiles_list_dup_uuid_or_name (terminal_options_ensure_profiles_list (options),
                                                      value, error);
+
+  if (value && profile == NULL)
+  {
+      g_warning ("Profile '%s' specified but not found. Attempting to fall back "
+              "to the default profile.", value);
+      g_clear_error (error);
+      profile = terminal_profiles_list_dup_uuid_or_name (terminal_options_ensure_profiles_list (options),
+                                                         NULL, error);
+  }
+
   if (profile == NULL)
     return FALSE;
 
